sweetest/src/main/java/com/mysugr/sweetest/framework/context/DependenciesTestContext.kt

```diff
@@ -83,7 +99,46 @@ class DependenciesTestContext {
      *  [provide]. [provide] automatically requires that the automatically created instance is used.
      */
     fun provide(clazz: KClass<*>) {
-        val dependency = getDependencyConfiguration(clazz)
-        TestEnvironment.dependencies.states[dependency].mode = DependencyMode.REAL
+        val mode = DependencyMode.REAL
+        addDependency(clazz) { state, _ ->
+            state.mode = mode
+        }
+    }
+
+    /**
+     * Checks if the type [clazz] is already added via the global module configuration.
+     * If so, the found [DependencyConfiguration] is added to the [DependencyManager.states].
+     * If not, a [DependencyConfiguration] is created dynamically and also added.
+     */
+    private fun addDependency(clazz: KClass<*>, block: (DependencyState<out Any>, DependencyMode?) -> Unit) {
+        with(TestEnvironment.dependencies) {
+            assertDependency(clazz)
+            val dependencyState = getDependencyConfiguration(clazz)?.let { configuration ->
+                states[configuration]
+            } ?: run {
+                states[DependencyConfiguration(clazz, null, null, null)]
+            }
+
+            block(dependencyState, dependencyState.configuration.defaultDependencyMode)
+        }
+    }
+
+    /**
+     * Asserts if the type [clazz] is already added.
+     */
+    private fun assertDependency(clazz: KClass<*>) {
```

Comment:
this check causes a lot of tests to fail in production when i tried it. we should further investigate this and add tests in sweetest for those circumstances...